source/slang/slang-ir-autodiff-unzip.h

```diff
@@ -372,10 +372,30 @@ struct DiffUnzipPass
             }
             else
             {
-                // For non differentiable arguments, we can simply pass the argument as is
-                // if this isn't a `out` parameter, in which case it is removed from propagate call.
-                if (!as<IROutType>(arg->getDataType()))
-                    diffArgs.add(arg);
+                // Ignore `out` parameters since these are removed from propagate call.
+                if (!as<IROutType>(resolvedPrimalFuncType->getParamType(ii)))
```

Comment:
General code style: avoid unnecessary nesting levels if possible. In this case, we can use three flat if-else cases.